src/pymmcore_plus/mda/_engine.py

```diff
@@ -149,6 +149,14 @@ def exec_event(self, event: MDAEvent) -> Iterable[PImagePayload]:
                 # store correction for this position index
                 p_idx = event.index.get("p", None)
                 self._z_correction[p_idx] = new_correction
+
+            # this is necessary for SequencedEvent, if we don't yield
+            # from self.exec_sequenced_event(event) the acquisition stops
+            # because we return ()
+            if isinstance(event, SequencedEvent):
```

Comment:
at first glance, i find it a bit odd to be checking `if isinstance(event, SequencedEvent):` inside of this `HardwareAutofocus` conditional, given that we check it again immediately below, and perform the same action `yield from self.exec_sequenced_event(event)`.  So it kinda feels to me like the `if` blocks themselves need evaluation.

One question I have is: "what is the meaning of a `SequencedEvent` that has a `HardwareAutofocus)` action?".  That is, why do we ever get here in the first place?  It's possible that we need to re-examine the building of a `SequenceEvent` and special case other actions...  Where you able to see more about the event that caused this?  Was it in fact a multi-frame event that also had a hardware action?